src/pages/en/guides/client-side-scripts.md

```diff
@@ -0,0 +1,205 @@
+
+```astro title="src/components/InlineScript.astro" "is:inline"
+<script is:inline>
+  // Will be rendered into the HTML exactly as written!
+  // ESM imports will not be resolved relative to the file.
+</script>
+```
+
+:::note
+Adding `type="module"` or any other attribute to a `<script>` tag will disable Astro's default bundling behavior, treating the tag as if it had an `is:inline` directive.
+:::
+
+📚 See our [directives reference](/en/reference/directives-reference/#script--style-directives) page for more information about the directives available on `<script>` tags.
+
+### Script loading
+
+You may want to write your scripts as separate `.js`/`.ts` files or need to reference an external script on another server. You can do this by referencing these in a `<script>` tag’s `src` attribute.
+
+#### Import local scripts
+
+**When to use this:** If your script lives inside of `src/`.
+
+Astro will build, optimize, and add these scripts to the page for you, following its [script bundling rules](#script-bundling).
+
+```astro title="src/components/LocalScripts.astro"
+<!-- relative path to script in `src/` -->
+<script src="./local-script.js"></script>
+
+<!-- also works for local TypeScript files -->
+<script src="./script-with-types.ts"></script>
+```
+
+#### Load external scripts
+
+**When to use this:** If your JavaScript file lives inside of `public/` or on a CDN.
+
+This approach skips the JavaScript processing, bundling, and optimizations that are provided by Astro when you import scripts as described above.
+
+```astro title="src/components/ExternalScripts.astro" "is:inline"
+<!-- absolute path to a script at `public/my-script.js` in your project -->
+<script is:inline src="/my-script.js"></script>
+
+<!-- full URL to a script on a remote server -->
+<script is:inline src="https://analytics.example.com/script.js"></script>
+```
+
+## Common script patterns
+
+### Handle `onclick` and other events
+
+Some UI frameworks use custom syntax for event handling like `onclick={...}` (React/Preact) or `@click="..."` (Vue). Astro components follow web standards and recommend using [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) in a `<script>` tag to handle user interactions.
+
+```astro title="src/components/AlertButton.astro"
+<button data-alert>Click me!</button>
+
+<script>
+  // Find all buttons with the `data-alert` attribute on the page.
+  const buttons = document.querySelectorAll('button[data-alert]');
+
+  // Handle clicks on each button.
+  buttons.forEach((button) => {
+    button.addEventListener('click', () => {
+      alert('Button was clicked!');
+    });
+  });
+</script>
+```
+
+:::note
+Because this script will run once even if we add many instances of `<AlertButton />` to a page, we use `querySelectorAll` to get all instances of our component and add listeners to each one instead of using `querySelector`, which would only return the first button it found.
```

Comment:
small suggestion, feel free to ignore
```suggestion
Because this script will be shipped only once even if we add many instances of `<AlertButton />` to a page, we use `querySelectorAll` to get all instances of our component and add listeners to each one instead of using `querySelector`, which would only return the first button it found.
```

Reply:
Oops, this got overridden by Sarah’s rewrite here. The new wording is “Scripts are bundled and run only once.” Do you think “bundled” is enough to convey the same concept as “shipped” here?